sycl/source/detail/device_binary_image.cpp

```diff
@@ -77,14 +77,14 @@ ByteArray DeviceBinaryProperty::asByteArray() const {
   return {Data, Prop->ValSize};
 }
 
-const char *DeviceBinaryProperty::asCString() const {
+std::string_view DeviceBinaryProperty::asStringView() const {
   assert((Prop->Type == SYCL_PROPERTY_TYPE_STRING ||
           Prop->Type == SYCL_PROPERTY_TYPE_BYTE_ARRAY) &&
          "property type mismatch");
   assert(Prop->ValSize > 0 && "property size mismatch");
   // Byte array stores its size in first 8 bytes
   size_t Shift = Prop->Type == SYCL_PROPERTY_TYPE_BYTE_ARRAY ? 8 : 0;
-  return ur::cast<const char *>(Prop->ValAddr) + Shift;
+  return {ur::cast<const char *>(Prop->ValAddr) + Shift, Prop->ValSize - Shift};
```

Comment:
Unrelated nitpicking, but this cast has nothing to do with `ur` and a standard C++ cast would fit better, I think.